### ISSUE TITLE
Feature request: If provided intermediate files within workflow, ability to skip previous steps

### ISSUE THREAD

**Issue by sooheelee** [openwdl/wdl#167]
This would greatly reduce my need to modify WDL scripts to start where I have data already processed. For example, if a script goes BAM-->coverage-->CNVs, if I have already collected coverage on my BAMs, I would like to be able to provide `coverage` to the same script and have Cromwell skip the tasks involving the BAM and run the remaining steps in the workflow, e.g. coverage-->CNVs. 

Let me know if I should have posted this to the WDL forum instead. Thanks.

**Comment by cbirger** [comment]
Hi Soo Hee,

Is it correct that what you describe is distinct from call caching?  In call caching, if a workflow task as already been run with the same inputs, then rather than re-running that task, the cached output from the earlier run will be used.  What you are describing, I believe, is an alternative entry point for a workflow, one that begins further along in the workflow, skipping the upstream tasks.  Could a WDL currently be designed to do this through judicious use of optional input parameters?

-Chet

**Comment by sooheelee** [comment]
I run WDLs using gcloud, within a VM and locally. I don't use FireCloud so my runs do not use call-caching.

I want to take the boilerplate WDL scripts the GATK4 repo makes available to run processes, currently the latest somatic CNV workflow. If I have alreaded padded my intervals and/or collected counts on from the BAMs, I'd like to still use the rest of the steps in the workflow by specifying in the INPUTS JSON an intermediate file.

If the script is thus:
```
   call CNVTasks.PreprocessIntervals {
        input:
            intervals = intervals,
            ref_fasta_dict = ref_fasta_dict,
            gatk4_jar_override = gatk4_jar_override,
            gatk_docker = gatk_docker
    }

    if (select_first([do_explicit_gc_correction, false])) {
        call CNVTasks.AnnotateIntervals {
            input:
                intervals = PreprocessIntervals.preprocessed_intervals,
                ref_fasta = ref_fasta,
                ref_fasta_fai = ref_fasta_fai,
                ref_fasta_dict = ref_fasta_dict,
                gatk4_jar_override = gatk4_jar_override,
                gatk_docker = gatk_docker
        }
```

In the inputs, instead of defining:
```
"CNVSomaticPanelWorkflow.intervals": "File",
```
I would like to be able to instead provide:
```
"CNVSomaticPanelWorkflow.iPreprocessIntervals.preprocessed_intervals": "File",
```

And not have the run error due to the lack of the `CNVSomaticPanelWorkflow.intervals` file.

**Comment by geoffjentry** [comment]
Hey - just a note that concepts like call caching and such are implementation details (e.g. Cromwell) and not WDL. @sooheelee you're not necessarily asking about caching but it's still regarding Cromwell and/or how a WDL implementation might optimize runtime. 

tl;dr You should file this over in the Cromwell repo

**Comment by sooheelee** [comment]
Have posted this to <https://github.com/broadinstitute/cromwell/issues/2949>. Thanks for pointing out the Cromwell repo!
